### ISSUE TITLE
Provisioning Profiles are Invalidated & Entitlements Option is Reset

### ISSUE THREAD

**Issue by rickmanelius** [bitrise-steplib/steps-ios-auto-provision-appstoreconnect#20]
So 1st off, my goal is not to have Bitrise.io automatically manage the provisioning profile, but I'm cool with it as long as it's functional. Right now, it's not. We are attempting to move from the legacy `ios-auto-provision` v1.5.4 step to this one, but we noticed some funny behavior. Not only is it creating a new provisioning profile with a slight difference in title vs lowercase (e.g. we had "Bitrise ios development - (com.contactmapping.mobile)" and now we have another as "Bitrise iOS development - (com.contactmapping.mobile)" with iOS vs ios), but we are also seeing the notes entitlement continually get reset and removed, causing the step to fail.

It looks like there might be a syntax problem. In [build 3566](https://app.bitrise.io/build/b412d2ca0d237e6c#?tab=log), I see the following:

```
  Checking bundle id: com.mycompany.mobile
  capabilities: map[aps-environment:development com.apple.developer.contacts.notes:%!s(bool=true)]
  Bitrise managed profile found: Bitrise iOS app-store - (com.mycompany.mobile)
  the profile is not in sync with the project requirements, regenerating ...
```

The `%!s` looks like it's not properly getting passed in. It tries to regenerate twice and continues without the entitlement.

To be clear, here's what is noted from the other version of the step.

```
Entitlements:

{
    "aps-environment" = development;
    "com.apple.developer.contacts.notes" = 1;
```

TL;DR. I'd like to see if this boolean issue is causing the multiple failed attempts to regenerate the profile.

Thanks

**Comment by mjones-droid** [comment]
Thanks for reporting. I've filed a bug report with the dev's.

**Comment by rickmanelius** [comment]
@mjones-droid Much appreciated!

**Comment by rickmanelius** [comment]
Hmm. On additional inspection, it might be that the log output is having difficulty converting the `entitlements serialized.Object` parameter to a %s output. However, the upstream issue of it regenerating despite the entitlements already matching what is noted in the provisioning profile is tricky.

**Comment by mjones-droid** [comment]
Hey @rickmanelius. Here's the official response from the dev's. 

> We can not fix this issue due to technical limitations, if that specific entitlement is needed they must use the Certificates and profiles installer step only to install certs and profiles (manually generated on Developer Portal)
> 
> The API does not allow adding entitlements as far as we know, only those that are covered by a Capability (Push Notification, …
> 
> We are adding an error to the autoprov appstoreconnect step if an unsupported entitlement is detected.

**Comment by rickmanelius** [comment]
Hi @mjones-droid. Perfect timing! I was just about to pick this one up again. Thanks for the tip on the `certificate-and-profile-installer`. It looks like I should have been attempting to use that one from the beginning instead of this one after we decided to manually manage certs and profiles. I'll close these two tickets and switch to that step. Thanks!
